web/docs/language/features.md

```diff
@@ -178,138 +178,246 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+  return context.entities.Task.findMany({
+    where: { isDone: args.isDone }
+  })
+}
+```
+
+The object `context.entities.Task` exposes `prisma.task` from [Prisma's CRUD API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
+
+
+### Action
+
+Actions are very similar to Queries. So similar, in fact, we will only list the differences:
+1. They can (and most often should) modify the server's state, while Queries are only allowed to read it.
+2. Since Actions don't need to be reactive, Wasp doesn't provide a React hook for them (like `useQuery` for Queries) - you just call them directly.
+3. `action` declarations in Wasp are mostly identical to `query` declarations. The only difference is in the declaration's name.
+
+Here's an implementation of a simple Action:
+
+```js title=actions.js
+export const sayHi = async () => {
+  console.log('The client said Hi!')
+}
 ```
+Its corresponding declaration in Wasp:
 
-where `context.entities.Task` actually exposes `prisma.task` from [Prisma API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
+```c title="main.wasp"
+// ...
 
-#### Cache invalidation
-One of the trickiest part of managing web app state is making sure that data which queries are showing is up to date.
+action sayHi {
+  fn: import { sayHi } from "@ext/actions.js"
+}
+```
+And an example of how to import and call the declared Action:
 
-Since Wasp is using react-query for managing queries, that means we want to make sure that parts of react-query cache are invalidated when we know they are not up to date any more.
+```js
+import sayHi from '@wasp/actions/sayHi'
 
-This can be done manually, by using mechanisms provided by react-query (refetch, direct invalidation).
-However, that can often be tricky and error-prone, so Wasp offers quick and effective solution to get you started: automatic invalidation of query cache based on entities that queries / actions are using.
+// ...
 
-Specifically, if Action A1 uses Entity E1 and Query Q1 also uses Entity E1 and Action A1 is executed, Wasp will recognize that Q1 might not be up-to-date any more and will therefore invalidate its cache, making sure it gets updated.
+sayHi()
+```
 
-In practice, this means that without really even thinking about it, Wasp will make sure to keep the queries up to date for you in regard with the changes done by actions.
 
-On the other hand, this kind of automatic invalidation of cache can be wasteful (updating when not needed) and will not work if other resources than entities are used. In that case, make sure to use mechanisms provided by react-query for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
+More differences and Action/Query specific features will come in future versions of Wasp.
 
+### Cache Invalidation
+One of the trickiest parts of managing a web app's state is making sure the data shown by the queries is up to date.
```

Comment:
```suggestion
One of the trickiest parts of managing a web app's state is making sure the data returned by the queries is up to date.
```
Maybe a bit more technically correct? Since e.g. React does the showing part.